### PR TITLE
[backend/frontend] Fix team creation

### PR DESCRIPTION
**Pull request by RomuDeuxfois**
### Related issues

* https://github.com/OpenBAS-Platform/openbas/issues/1398

### CODE REVIEW

File: openbas-model/src/main/java/io/openbas/database/repository/TeamRepository.java [hunk 1]
```diff
@@ -28,6 +28,9 @@ public interface TeamRepository extends CrudRepository<Team, String>,
   @NotNull
   Optional<Team> findByName(@NotNull final String name);
 
+  @NotNull
+  List<Team> findAllByName(@NotNull final String name);
+
```
Comment (savacano28) [inline openbas-model/src/main/java/io/openbas/database/repository/TeamRepository.java:33]: Should we use ignoreCase, or would it be better to create a task to rearrange this from the creation?

File: openbas-api/src/main/java/io/openbas/rest/team/TeamApi.java [hunk 2]
```diff
@@ -236,4 +232,26 @@ public Team updateTeamUsers(@PathVariable String teamId, @Valid @RequestBody Upd
...
+        List<Team> teams = this.teamRepository.findAllByName(input.getName());
+        // 1. input is not contextual and a not contextual team already exist
+        if (FALSE.equals(input.getContextual()) && teams.stream().anyMatch(t -> FALSE.equals(t.getContextual()))) {
+            throw new UnsupportedOperationException("Global teams (non contextual) cannot have the same name (already exists)");
+        }
+        // 2. input is contextual and a contextual team on the same exercise or scenario already exist
+        if (TRUE.equals(input.getContextual())) {
+            String exerciseId = input.getExerciseIds().stream().findFirst().orElse(null);
+            if (hasText(exerciseId) && teams.stream().anyMatch(t -> TRUE.equals(t.getContextual()) && t.getExercises().stream().anyMatch((e) -> exerciseId.equals(e.getId())))) {
+                throw new UnsupportedOperationException("A contextual team with the same name already exists on this simulation");
```
Comment (Dimfacion) [inline openbas-api/src/main/java/io/openbas/rest/team/TeamApi.java:248]: Maybe a stupid question but how come we can't have 2 teams with the same name ? Is it a rule required from product ?

File: openbas-model/src/main/java/io/openbas/database/repository/TeamRepository.java (hunk 1 above)
Comment (RomuDeuxfois) [reply openbas-model/src/main/java/io/openbas/database/repository/TeamRepository.java:33]: Good question, what do you think @jborozco & @EllynBsc ?
Could we have two teams like that: TeamA & tEAMa ?
Or not ?

File: openbas-api/src/main/java/io/openbas/rest/team/TeamApi.java (hunk 2 above)
Comment (RomuDeuxfois) [reply openbas-api/src/main/java/io/openbas/rest/team/TeamApi.java:248]: Good question, what do you think @jborozco & @EllynBsc  ?
Could we have two teams with the same name: TeamA & TeamA ?

Comment (EllynBsc) [reply openbas-api/src/main/java/io/openbas/rest/team/TeamApi.java:248]: Hey @RomuDeuxfois & @Dimfacion 👋 
I don't see the usecase of having two teams with the same name IMO. I think differentiated naming gives structures and fosters clarity.

File: openbas-api/src/main/java/io/openbas/rest/team/TeamApi.java [hunk 3]
```diff
@@ -236,4 +232,26 @@ public Team updateTeamUsers(@PathVariable String teamId, @Valid @RequestBody Upd
...
+            }
+            String scenarioId = input.getScenarioIds().stream().findFirst().orElse(null);
+            if (hasText(scenarioId) && teams.stream().anyMatch(t -> TRUE.equals(t.getContextual()) && t.getScenarios().stream().anyMatch((e) -> scenarioId.equals(e.getId())))) {
+                throw new UnsupportedOperationException("A contextual team with the same name already exists on this simulation");
+            }
```
Comment (savacano28) [inline openbas-api/src/main/java/io/openbas/rest/team/TeamApi.java:253]: "on this scenario"

File: openbas-api/src/main/java/io/openbas/rest/team/TeamApi.java [hunk 4]
```diff
@@ -236,4 +232,26 @@ public Team updateTeamUsers(@PathVariable String teamId, @Valid @RequestBody Upd
         team.setUsers(fromIterable(teamUsers));
         return teamRepository.save(team);
     }
+
+    // -- PRIVATE --
+
+    private void isTeamAlreadyExists(@NotNull final TeamCreateInput input) {
+        List<Team> teams = this.teamRepository.findAllByName(input.getName());
+        // 1. input is not contextual and a not contextual team already exist
```
Comment (savacano28) [inline openbas-api/src/main/java/io/openbas/rest/team/TeamApi.java:240]: If the teams are empty, we could make: directly a return; ?

File: openbas-api/src/main/java/io/openbas/rest/team/TeamApi.java (hunk 3 above)
Comment (savacano28) [reply openbas-api/src/main/java/io/openbas/rest/team/TeamApi.java:253]: Maybe manage better this unsopportedOperationException (500 ->400, 409?). 
On the frontend, we have: 
image

**Comment by RomuDeuxfois** [comment]
Add tests and show error on front

File: openbas-model/src/main/java/io/openbas/database/repository/TeamRepository.java (hunk 1 above)
Comment (RomuDeuxfois) [reply openbas-model/src/main/java/io/openbas/database/repository/TeamRepository.java:33]: Adding ignoreCase

**Review by savacano28 — APPROVED** [review]
Tested ok

**Pull request merged**